### PR TITLE
Restore recognition of deprecated `FIDESCTL__API__` ENV variables

### LINKED ISSUE #963: Env variable not overriding database host config in 1.8.0
### Bug Description

`FIDESCTL__API__DATABASE_HOST` no longer works to override the toml config.

#### Steps to Reproduce

1. `docker run -it -e FIDESCTL__API__DATABASE_HOST=foo ethyca/fidesctl:1.8.0`
    This will give an error

    ```
    2022-08-04 21:28:20.712 [ERROR]: Unable to configure database: sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) could not translate host name "fidesctl-db" to address: Name or service not known
    ```

    This should be `foo`, not `fidesctl-db`

1. `docker run -it -e FIDESCTL__DATABASE__SERVER=foo ethyca/fidesctl:1.8.0`
    This one does work, as it gives an error about not being able to connect to `foo`
    ```
    2022-08-04 21:36:29.821 [ERROR]: Unable to configure database: sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) could not translate host name "foo" to address: No address associated with hostname
    ```

### Expected behavior

Should connect to the database host based on the environment variable if it exists.

### Screenshots

_If applicable, add screenshots to help explain your problem._

### Environment

* Version: 1.8.0
* OS:
* Python Version:
* Docker Version:

### PR DESCRIPTION
**Pull request by PSalant726**
Closes #963

### Code Changes

* [ ] When generating the `config` object, search the local ENV for variables prefixed with `FIDESCTL__API__`, and set any corresponding `[database]` config values

### Steps to Confirm

* [ ] Set an old ENV variable (i.e. `FIDESCTL__API__DATABASE_HOST`)
* [ ] Start the server
* [ ] Confirm that `config.database.server` is set to the value of `FIDESCTL__API__DATABASE_HOST`

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation Updated:
  - [x] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] ~documentation issue created (tag docs-team to complete issue separately)~
* [x] Issue Requirements are Met
* [ ] ~Relevant Follow-Up Issues Created~
* [x] Update `CHANGELOG.md`

### CODE REVIEW

File: tests/ctl/core/test_config.py [hunk 1]
```diff
@@ -46,6 +46,34 @@ def test_get_deprecated_api_config(test_deprecated_config_path: str) -> None:
...
 
+@patch.dict(
+    os.environ,
+    {
+        "FIDESCTL__API__DATABASE_HOST": "test_host",
+        "FIDESCTL__API__DATABASE_NAME": "test_db_name",
+        "FIDESCTL__API__DATABASE_PASSWORD": "test_password",
+        "FIDESCTL__API__DATABASE_PORT": "1234",
+        "FIDESCTL__API__DATABASE_TEST_DATABASE_NAME": "test_test_db_name",
+        "FIDESCTL__API__DATABASE_USER": "phil_rules",
```
Comment (NevilleS) [inline tests/ctl/core/test_config.py:57]: 👀

Full disclosure, I went straight to the unit tests, so I'm glad to see these here

**Review by NevilleS — APPROVED** [review]
LGTM, but you should get one more review from a core team member who groks the style better 👍

Thanks

File: src/fidesctl/ctl/core/config/__init__.py [hunk 2]
```diff
@@ -49,6 +51,30 @@ def handle_deprecated_fields(settings: MutableMapping) -> MutableMapping:
     return settings
 
 
+def handle_deprecated_env_variables(settings: MutableMapping) -> MutableMapping:
+    """
+    Custom logic for handling deprecated ENV variable configuration.
+    """
+
+    deprecated_env_vars = regex(r"FIDESCTL__API__(\w+)")
```
Comment (PSalant726) [inline src/fidesctl/ctl/core/config/__init__.py:59]: @ThomasLaPiana I believe this should cover all of the deprecated ENV variables? We can always add more like:
```py
deprecated_env_vars = regex(r"FIDESCTL__[API|FOO|BAR]__(\w+)")
```

Comment (ThomasLaPiana) [reply src/fidesctl/ctl/core/config/__init__.py:59]: This covers all of them, thanks! 👍

**Review by ThomasLaPiana — APPROVED** [review]
LGTM! great fix

**Pull request merged**